app/models/jobstatusdata.rb

```diff
@@ -7,77 +7,146 @@
 # @version 0.0.1
 class Jobstatusdata
 
-  attr_reader :pbsid, :jobname, :username, :account, :status, :cluster, :nodes, :starttime, :walltime, :walltime_used, :submit_args, :output_path, :nodect, :ppn, :total_cpu, :queue, :cput, :mem, :vmem, :terminal_path, :fs_path
+  attr_reader :pbsid, :jobname, :username, :account, :status, :cluster, :nodes, :starttime, :walltime, :walltime_used, :submit_args, :output_path, :nodect, :ppn, :total_cpu, :queue, :cput, :mem, :vmem, :terminal_path, :fs_path, :extended_available
 
   # Define an object containing only necessary data to send to client.
   #
   # Object defaults to condensed data, add extended flag to initializer to include all data used by the application.
   #
   # @return [Jobstatusdata] self
-  def initialize(pbs_job, pbs_cluster=Servers.first[0], extended=false)
-    self.pbsid = pbs_job[:name]
-    self.jobname = pbs_job[:attribs][:Job_Name]
-    self.username = username_format(pbs_job[:attribs][:Job_Owner])
-    self.account = pbs_job[:attribs][:Account_Name]
-    self.status = pbs_job[:attribs][:job_state]
-    if self.status == "R" || self.status == "C"
-      self.nodes = node_array(pbs_job[:attribs][:exec_host])
-      self.starttime = pbs_job[:attribs][:start_time]
+  def initialize(info, cluster=OODClusters.first.id.to_s, extended=false)
+    self.pbsid = info.id
+    self.jobname = info.job_name
+    self.username = info.job_owner
+    self.account = info.accounting_id
+    self.status = info.status.state
+    self.cluster = cluster
+    self.walltime_used = info.wallclock_time > 0 ? pretty_time(info.wallclock_time) : ''
+    self.queue = info.queue_name
+    if info.status == :running || info.status == :completed
+      self.nodes = node_array(info.allocated_nodes)
+      self.starttime = info.dispatch_time.to_i || 0
```

Comment:
```
2.2.0 :002 > nil.to_i
 => 0
```

So perhaps for this `info.dispatch_time.to_i` is okay without the guard clause because if info.dispatch_time is nil then it will be 0